### PR TITLE
Add config docs

### PR DESCRIPTION
**Pull request by Dr-Electron**
# Description of change

Added docs for the config

## Type of change

- Documentation Fix

## Change checklist

- [x ] I have followed the contribution guidelines for this project

### CODE REVIEW

**Comment by Dr-Electron** [comment]
Needs some refactoring. Working on it. Will try to finish tomorrow.

**Comment by Dr-Electron** [comment]
A nice. We already have tomorrow ;). PR open for review again

File: docs/configuration.md [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+---
+
+| Name       | Description                                                                                | Type   |
+| :--------- | :----------------------------------------------------------------------------------------- | :----- |
+| identity   | hex representation of an Ed25519 keypair. Can be generated with the `bee p2pidentity` tool | string |
+| alias      | alias for your node. Shows up in dashboard                                                 | string |
+| bech32_hrp | network address identifier                                                                 | string |
+| network_id | network identifier                                                                         | string |
+
+## 1. Logger
```
Comment (thibault-martinez) [inline docs/configuration.md]: I wouldn't number the categories, doesn't really add anything and makes it annoying if/when we reorder them

File: docs/configuration.md [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+| target_width        | width of the target section of a log | integer[usize] |
+| level_width         | width of the level section of a log  | integer[usize] |
+| [outputs](#outputs) | config for different log filters     | array          |
+
+### outputs
+
+| Name         | Description             | Type   |
+| :----------- | :---------------------- | :----- |
+| name         | standart stream or file | string |
+| level_filter | log level filter        | string |
```
Comment (thibault-martinez) [inline docs/configuration.md]: There is also target fiter

File: docs/configuration.md [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+address = "/dns/example.com/tcp/15600/p2p/PeerID"
+alias   = "yet another peer"
+```
+
+## 3. Protocol
+
+| Name                        | Description           | Type         |
+| :-------------------------- | :-------------------- | :----------- |
+| minimum_pow_score           | the minimum pow score | float[f64]   |
+| handshake_window            | TO-DO                 | integer[u64] |
```
Comment (thibault-martinez) [inline docs/configuration.md]: Old one, remove

File: docs/configuration.md [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+public_key  = "0758028d34508079ba1f223907ac3bb5ce8f6bdccc6b961c7c85a2f460b30c1d"
+start       = 0
+end         = 0
+[protocol.workers]
+message_worker_cache = 10000
+status_interval = 10
+ms_sync_count = 200
+```
+
+## 4. Rest_api
```
Comment (thibault-martinez) [inline docs/configuration.md]:
```suggestion
## 4. REST API
```

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: docs/configuration.md (hunk 1 above)
Comment (Dr-Electron) [reply docs/configuration.md]: You are right. Problem with that is always, that sometimes you don't have a unique identifier anymore. For example with `[storage]` and `[storage.storage]`

Comment (thibault-martinez) [reply docs/configuration.md]: That's just an issue for the anchor, right ? Do we really care since it's not shown ?

Comment (Dr-Electron) [reply docs/configuration.md]: The problem is, that in the ToC I need to link to the storage table and the other storage table in the storage table. How should I do that, if the both have the same name? Or do I don't get something?

File: docs/configuration.md [hunk 5]
```diff
@@ -0,0 +1,394 @@
...
+public_key  = "0758028d34508079ba1f223907ac3bb5ce8f6bdccc6b961c7c85a2f460b30c1d"
+start       = 0
+end         = 0
+[protocol.workers]
+message_worker_cache = 10000
+status_interval = 10
+ms_sync_count = 200
+```
+
+## Rest_api
```
Comment (thibault-martinez) [inline docs/configuration.md]: REST API

File: docs/configuration.md [hunk 6]
```diff
@@ -0,0 +1,394 @@
...
+
+[//]: # "Table of contents created with: http://ecotrust-canada.github.io/markdown-toc/"
+
+- [Logger](#logger)
+  * [outputs](#outputs)
+- [Network](#network)
+    + [peering](#peering)
+- [Protocol](#protocol)
+  * [coordinator](#coordinator)
+    + [public_key_ranges](#public-key-ranges)
```
Comment (thibault-martinez) [inline docs/configuration.md]: Link for this one is not working

File: docs/configuration.md (hunk 5 above)
Comment (Dr-Electron) [reply docs/configuration.md]: How did I revert that one? Will fix it again ;)

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**